server/aws-lsp-codewhisperer/src/language-server/codeWhispererServer.ts

```diff
@@ -526,26 +515,31 @@ export const CodewhispererServerFactory =
             emitUserTriggerDecisionTelemetry(telemetry, session, timeSinceLastUserModification)
         }
 
-        const updateConfiguration = async () =>
-            lsp.workspace
-                .getConfiguration('aws.codeWhisperer')
-                .then(config => {
-                    if (config && config['includeSuggestionsWithCodeReferences'] === true) {
-                        includeSuggestionsWithCodeReferences = true
-                        logging.log('Configuration updated to include suggestions with code references')
-                    } else {
-                        includeSuggestionsWithCodeReferences = false
-                        logging.log('Configuration updated to exclude suggestions with code references')
-                    }
-                    if (config && config['shareCodeWhispererContentWithAWS'] === true) {
-                        codeWhispererService.shareCodeWhispererContentWithAWS = true
-                        logging.log('Configuration updated to share Amazon Q content with AWS')
-                    } else {
-                        codeWhispererService.shareCodeWhispererContentWithAWS = false
-                        logging.log('Configuration updated to not share Amazon Q content with AWS')
-                    }
-                })
-                .catch(reason => logging.log(`Error in GetConfiguration: ${reason}`))
+        const updateConfiguration = async () => {
+            try {
+                const qConfig = await lsp.workspace.getConfiguration(Q_CONFIGURATION_SECTION)
+                if (qConfig) {
+                    codeWhispererService.customizationArn = qConfig.customization ?? ''
+                    logging.log(`Inline completion configuration updated to use ${qConfig.customization}`)
+                }
+
+                const config = await lsp.workspace.getConfiguration('aws.codeWhisperer')
```

Comment:
nit: it should be possible to request multiple configuration sections with one `getConfiguration` call, instead of 2 https://github.com/microsoft/vscode-languageserver-node/blob/f58f4dff16ad2760028bb1cb95e882de30a1000f/server/src/common/configuration.ts#L18.

Our `Workspace.getConfiguration` proxy handler only supports `section` though, so it will need to be extended. Could be done as a follow-up.